pkg/kubectl/resource/builder.go

```diff
@@ -440,15 +440,27 @@ func (b *Builder) resourceMappings() ([]*meta.RESTMapping, error) {
 	}
 	mappings := []*meta.RESTMapping{}
 	for _, r := range b.resources {
-		gvk, err := b.mapper.KindFor(unversioned.GroupVersionResource{Resource: r})
+		gvks, err := b.mapper.KindsFor(unversioned.GroupVersionResource{Resource: r})
 		if err != nil {
 			return nil, err
 		}
-		mapping, err := b.mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
-		if err != nil {
-			return nil, err
+		// the list is in most-preferred to least preferred, so iterate until we find a hit
+		var firstErr error
```

Comment:
Probably this should be abstracted into a method `getFirstMapping()` and reused here and below.